src/components/ConfirmLeavingPage/ConfirmLeavingPage.test.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { PropsWithChildren } from 'react';
+import { Router } from 'react-router-dom';
+import { CompatRouter, Route, Routes } from 'react-router-dom-v5-compat';
+import { locationService } from '@grafana/runtime';
+import { TextLink } from '@grafana/ui';
+import { fireEvent, render, screen } from '@testing-library/react';
+import userEventLib from '@testing-library/user-event';
+
+import { DataTestIds } from '../../test/dataTestIds';
+import { ConfirmLeavingPage } from './ConfirmLeavingPage';
+
+const TEST_IDS = {
+  INITIAL_PAGE: 'ConfirmLeavingPage.initial-page',
+  LEAVE_PAGE_LINK: 'ConfirmLeavingPage.leave-page-link',
+  OTHER_PAGE: 'ConfirmLeavingPage.other-route',
+} as const;
+
+function Wrapper({ children }: PropsWithChildren<{}>) {
+  const history = locationService.getHistory();
+  // History will not automatically be reset between tests
+  history.replace('/');
```

Comment:
I'm worried this is easy to miss when maintaining this test in the future and it doesn't follow the convention of being in a `beforeEach` or similar where I'd expect to see this resetting behaviour.

Maybe remove this and add to the start of the `describe` block:

```
  beforeEach(() => {
    // History needs to be reset manually between tests as it uses Grafana's global locationService
    locationService.replace('/');
  });
```